Better constructors for rate_helpers

thrasibule opened the pull request:
allow to pass a float instead of a Quote
- for SwapRateHelper, the code was intended to work like that, but
  wasn't working
- also fixed a bug in DepositRateHelper where end_of_month was never
  used

dpinte commented on quantlib/termstructures/yields/rate_helpers.pyx:
```diff
@@ -71,12 +72,18 @@ cdef class RelativeDateRateHelper:
+                new _qt.SimpleQuote(<Rate>quote)))
+        elif isinstance(quote, SimpleQuote):
+            rate_handle = Handle[_qt.Quote](deref((<SimpleQuote>quote)._thisptr))
+
```
you might want to raise a `ValueError` if the `quote` is neither a float or a SimpleQuote.

dpinte commented on quantlib/test/test_rate_helpers.py:
```diff
@@ -26,13 +26,19 @@ def test_create_deposit_rate_helper(self):
 
 
-        helper = DepositRateHelper(
+        helper1 = DepositRateHelper(
```
I would suggest avoiding cryptic variable names. You could use `helper_from_quote` and `helper_from_float` in place of `helper1` and `helper2`

thrasibule commented:
I updated it based on your comments. I also updated FuturesRateHelper and FRARateHelper on the same model to be consistent.
Also added latest_date method, and tests.

dpinte commented:
Thanks @thrasibule! LGTM
